sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -604,6 +606,24 @@ class XmlTokenizer(
         return Some(str)
       }
     } catch {
```

Comment:
Consider this to avoid reader.close repetitions:

```
  def next(): Option[String] = {
    var nextString: Option[String] = None
    try {
      if (readUntilStartElement()) {
        buffer.append(currentStartTag)
        // Don't check whether the end element was found. Even if not, return everything
        // that was read, which will invariably cause a parse error later
        readUntilEndElement(currentStartTag.endsWith(">"))
        nextString = buffer.toString()
        buffer = new StringBuilder()
      }
    } catch {
      case e: FileNotFoundException if options.ignoreMissingFiles =>
        logWarning(
          "Skipping the rest of" +
          " the content in the missing file during schema inference",
          e)
      case NonFatal(e) =>
        ExceptionUtils.getRootCause(e) match {
          case _: RuntimeException | _: IOException if options.ignoreCorruptFiles =>
            logWarning(
              "Skipping the rest of" +
              " the content in the corrupted file during schema inference",
              e
            )
        }
    } finally {
      if (var == None) {
        reader.close()
        reader = null
      }
    }
    nextStr
  }
```

Reply:
Thanks for putting this down! 👍